src/utils/settings.cpp

```diff
@@ -32,31 +34,56 @@ namespace mp = multipass;
 namespace
 {
 const auto file_extension = QStringLiteral("conf");
+const auto daemon_root = QStringLiteral("local");
 const auto petenv_name = QStringLiteral("primary");
+
 std::map<QString, QString> make_defaults()
+{ // clang-format off
+    return {{mp::petenv_key, petenv_name},
+            {mp::driver_key, mp::platform::default_driver()}};
+} // clang-format on
+
+/*
+ * We make up our own file names to:
+ *   a) avoid unknown org/domain in path;
+ *   b) write daemon config to a central location (rather than user-dependent)
+ * Examples:
+ *   - ${HOME}/.config/multipass/multipass.conf
+ *   - /root/.config/multipass/multipassd.conf
+ */
+QString file_for(const QString& key) // the key should have passed checks at this point
 {
-    return {{mp::petenv_key, petenv_name}};
+    static const auto file_path_base = QStringLiteral("%1/%2.%3"); // static consts ensure these stay fixed
+    static const auto client_dir_path = QStandardPaths::writableLocation(QStandardPaths::AppConfigLocation);
+    static const auto daemon_dir_path = QStringLiteral("/etc/xdg"); // temporary
```

Comment:
Oh and this, won't work on Windows or macOS, or when we move to a strict snap. We might need to make this a platform thing?

Reply:
Right, I meant to do that but forgot. On the strict snap though, won't it see its own filesystem /etc/xdg?

Reply:
No, even strict snaps see *a* rootfs, in their case it's the base (i.e. `core16`) in our case. They still can only write things in any of `$SNAP{,_USER}_{DATA,COMMON}` by default.

[Here](https://docs.snapcraft.io/environment-variables) is a handful reference to the environment variables.

Reply:
OK, so we need to move this and implement differently depending on the platform. But should I also use `SNAP_COMMON` on Linux if available? Or leave that to deal with separately in the confinement story?

BTW, the strict snap would still be able to see `/etc/xdg` but not write to it, is that it?

Reply:
Or maybe SNAP_DATA, to allow changing it between versions. I suppose those contents won't be lost from one version to the next?

Reply:
I think yes, using `$SNAP_COMMON` if it's set is the right approach.

For things that people should generally not see/edit, the better default place on Linux is `/var/lib/`.

We shouldn't spend too much though on this though, as it's going to go away once we push the configs through the RPC.

Reply:
Ah and about the strict snap, by default it's not going to be able to read /etc/xdg I don't think. The policies are whitelists, so unless there's something that all snaps should have access to, or we connect to an interface, it only has access to its own bits and data (and a handful of things from the base, like libc and whatever's shipped there).